backend/app/database/weaviate/client.py

```
@@ -15,18 +18,46 @@ def get_client():
+
+async def close_weaviate_client():
+    """Close the Weaviate client. Call this on application shutdown."""
+    global _client, _connected
+    
+    if _client is not None and _connected:
         try:
-            await client.close()
+            await _client.close()
+            _connected = False
+            logger.info("Weaviate client closed")
         except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)